scripts/generate_json_docs.py

```diff
@@ -296,9 +299,34 @@ def clean_source_path(module):
     return '%s.py' % (source_id.replace('.', '/'),)
 
 
+def get_examples_from_docstring(doc_str):
+    examples = doctest.DocTestParser().get_examples(doc_str)
+    example_str = ''
+    for example in examples:
+        example_str += '%s' % example.source
+        example_str += '%s' % example.want
+
+    return example_str.replace('<', '&lt;').replace('>', '&gt;')
+
+
+def get_source_line_number(module):
+    if isinstance(module, (types.ModuleType, types.ClassType,
+                           types.MethodType, types.FunctionType,
+                           types.TracebackType, types.FrameType,
+                           types.CodeType, types.TypeType)):
+
+        _, line = inspect.getsourcelines(module)
+        source_path = clean_source_path(module)
+
+        if line:
+            source_path = source_path + '#L' + str(line)
+        return source_path
+
+
 def process_code_blocks(doc):
     blocks = []
     index = 0
+
```

Comment:
My flake8 plugin complains if there isn't a newline at the end of a file...but it's been flakey and not running after every save lately...no pun intended.